lib/fuel/engine.rb

```diff
@@ -1,5 +1,17 @@
 module Fuel
   class Engine < ::Rails::Engine
-    isolate_namespace Fuel
+
+    # ISOLATE NAMESPACE ONLY FOR MODELS
+    mod = Fuel
+    engine_name(generate_railtie_name(mod))
+    unless mod.respond_to?(:railtie_namespace)
+      name, railtie = engine_name, self
+      mod.singleton_class.instance_eval do
+        unless mod.respond_to?(:table_name_prefix)
+          define_method(:table_name_prefix) { "#{name}_" }
+        end
+      end
+    end
```

Comment:
I basically hijacked the isolate_namespace method so that it doesn't isolate everything...Fuel needs access to the main app's routes and application controller, for example. However, we do want the table name to be namespaced; fuel_posts == Fuel::Post. 

http://apidock.com/rails/Rails/Engine/isolate_namespace/class